### DISCUSSION TITLE
Writing Computer Name As Text On Image File

### DISCUSSION THREAD

**Discussion by ALittleOfEverything88** [ImageMagick/ImageMagick#4949] [Help] [answered]
All,

I'm trying to figure out how to write the computer name of the Windows 10 computer that the ImageMagick command is run on to an image file.

I've been able to come up with this:
`magick.exe convert -fill white -gravity NorthEast -pointsize 50 -annotate +50+300 "System name:" Input_LockScreen.jpg Output_LockScreen.jpg`

It works well, however, I cannot figure out how to have the application enter in the computer name after "System name:".; the computer name would change based upon the computer the command is running on. I've tried both batch file and PowerShell without success.

Is there a way to add the host name / computer name / system name using this application? 

Thanks!

**Comment by snibgo** [comment, marked as answer]
In Windows 8.1, you can use the system variable "%COMPUTERNAME%". Perhaps Windows 10 has the same. So you would have:
```
... "System name: %COMPUTERNAME%" ...
```

**Reply by ALittleOfEverything88** [reply]
That worked, thanks! Is there anywhere I can find a list of variables (similar to the above) that work with the application?

**Comment by snibgo** [comment]
This works with any application, not just ImageMagick. The software that translates the environment variable is in the Windows CMD interpreter.

For a list of environment variable that are currently set, type "set" at the command line.

**Comment by ALittleOfEverything88** [comment]
For anyone who finds this in the future; the issue is that I was trying to run the commands from Powershell - this has to be run from a batch file.
